src/api/index.tsx

```diff
@@ -536,39 +550,57 @@ export class ApiClient {
       if (filter.tags !== undefined) {
         params.push(`tags=${filter.tags.join(",")}`);
       }
+      if (pageSize !== undefined) {
+        params.push(`page_size=${pageSize}`);
+      }
+
       if (params.length === 0) return "";
       return "?" + params.join("&");
     };
 
-    const queryString = buildIncidentsQuery(filter);
+    if (!cursor) {
+      const queryString = buildIncidentsQuery(filter, pageSize);
+
+      return resolveOrReject(
+        this.authGet<CursorPaginationResponse<Incident>, never>(`/api/v1/incidents/${queryString}`),
+        defaultResolver,
+        (error) => new Error(`Failed to get incidents: ${error}`),
+      );
+    } else {
+      return resolveOrReject(
+        this.authGet<CursorPaginationResponse<Incident>, never>(cursor),
+        defaultResolver,
+        (error) => new Error(`Failed to get incidents: ${error}`),
+      );
+    }
+  }
 
-    return resolveOrReject(
-      this.authGet<Incident[], never>(`/api/v1/incidents/${queryString}`),
-      defaultResolver,
-      (error) => new Error(`Failed to get incidents: ${error}`),
-    );
+  // NOTE: This won't actually get all incidents anymore because of the pagination
+  // TODO: Remove all use of this method.
+  public getAllIncidentsFiltered(filter: IncidentsFilterOptions): Promise<Incident[]> {
+    return this.getPaginatedIncidentsFiltered(filter, null).then(paginationResponseResolver);
   }
 
   public getAllIncidents(): Promise<Incident[]> {
     return resolveOrReject(
-      this.authGet<Incident[], never>(`/api/v1/incidents/`),
-      defaultResolver,
+      this.authGet<CursorPaginationResponse<Incident>, never>(`/api/v1/incidents/`),
+      paginationResponseResolver,
       (error) => new Error(`Failed to get incidents: ${error}`),
     );
   }
 
   public getOpenIncidents(): Promise<Incident[]> {
     return resolveOrReject(
-      this.authGet<Incident[], never>(`/api/v1/incidents/open/`),
-      defaultResolver,
+      this.authGet<CursorPaginationResponse<Incident>, never>(`/api/v1/incidents/open/`),
+      paginationResponseResolver,
       (error) => new Error(`Failed to get incidents: ${error}`),
     );
   }
 
   public getOpenUnAckedIncidents(): Promise<Incident[]> {
     return resolveOrReject(
-      this.authGet<Incident[], never>(`/api/v1/incidents/open+unacked/`),
-      defaultResolver,
+      this.authGet<CursorPaginationResponse<Incident>, never>(`/api/v1/incidents/open+unacked/`),
+      paginationResponseResolver,
       (error) => new Error(`Failed to get incidents: ${error}`),
     );
   }
```

Comment:
These two endpoints are no longer needed since you can filter on open and acked orthogonally, but let's not do that now. In 1.1 maybe.